### PR TITLE
[ENH] Coverage for benchmarking

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs

part of #664

#### What does this implement/fix? Explain your changes.

adds tests for benchmarking module, particularly the critical difference calculations.

### CODE REVIEW

File: aeon/benchmarking/tests/test_critical_difference.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+# copyright: aeon developers, BSD-3-Clause License (see LICENSE file)
```
Comment (chrisholder) [inline aeon/benchmarking/tests/test_critical_difference.py:2]: Is it normal to have copyright at the top of the files?

File: aeon/benchmarking/tests/test_critical_difference.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    nemenyi_cliques,
+    plot_critical_difference,
+    wilcoxon_holm_cliques,
+)
+from aeon.benchmarking.results_loaders import get_estimator_results_as_array
+from aeon.datasets.tsc_data_lists import univariate_equal_length
+from aeon.utils.validation._dependencies import _check_soft_dependencies
+
+test_path = MODULE = os.path.dirname(__file__)
+data_path = os.path.join(test_path, "../example_results/")
```
Comment (chrisholder) [inline aeon/benchmarking/tests/test_critical_difference.py:22]: These should probably all caps DATA_PATH and TEST_PATH 

Why are you assigning two constants the same value here:
`test_path = MODULE = os.path.dirname(__file__)`

File: aeon/benchmarking/tests/test_results_loaders.py [hunk 3]
```diff
@@ -14,11 +14,8 @@
 
 cls = ["HC2", "FreshPRINCE", "InceptionT"]
 data = ["Chinatown", "Tools"]
-data_path = (
-    "./aeon/benchmarking/example_results/"
-    if os.getcwd().split("\\")[-1] != "tests"
-    else "../example_results/"
-)
+test_path = MODULE = os.path.dirname(__file__)
```
Comment (chrisholder) [inline aeon/benchmarking/tests/test_results_loaders.py:17]: Same

File: aeon/benchmarking/tests/test_results_plotting.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+from aeon.benchmarking.results_loaders import get_estimator_results_as_array
+from aeon.benchmarking.results_plotting import (
+    plot_boxplot_median,
+    plot_scatter,
+    plot_scatter_predictions,
+)
+from aeon.datasets.tsc_data_lists import univariate_equal_length
+from aeon.utils.validation._dependencies import _check_soft_dependencies
+
+test_path = MODULE = os.path.dirname(__file__)
```
Comment (chrisholder) [inline aeon/benchmarking/tests/test_results_plotting.py:18]: Same

**Review by chrisholder — CHANGES_REQUESTED** [review]

File: aeon/benchmarking/tests/test_critical_difference.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/benchmarking/tests/test_critical_difference.py]: I think we have it some places, but dont think its normal

File: aeon/benchmarking/tests/test_critical_difference.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/benchmarking/tests/test_critical_difference.py:21]: god knows! :) fixed

File: aeon/benchmarking/tests/test_results_plotting.py (hunk 4 above)
Comment (TonyBagnall) [reply aeon/benchmarking/tests/test_results_plotting.py:18]: fixed

**Review by chrisholder — APPROVED** [review]

**Pull request merged**